src/WorkflowCore.DSL/Services/DefinitionLoader.cs

```diff
@@ -217,17 +217,76 @@ private void AttachOutputs(StepSourceV1 source, Type dataType, Type stepType, Wo
                 var sourceExpr = DynamicExpressionParser.ParseLambda(new[] { stepParameter }, typeof(object), output.Value);
 
                 var dataParameter = Expression.Parameter(dataType, "data");
-                Expression targetProperty;
+                Expression targetProperty = dataParameter;
 
                 // Check if our datatype has a matching property
-                var propertyInfo = dataType.GetProperty(output.Key);
-                if (propertyInfo != null)
+
+                PropertyInfo propertyInfo = null;
+                String[] paths = output.Key.Split('.');
+
+                bool hasAddOutput = false;
+
+                foreach (String propertyName in paths)
```

Comment:
This method is getting quite big and doing lots of things... could we split the nested property flattening into it's own method?

Reply:
ok, split the nested property and direct property using AttachNestedOutput AttachDirectlyOutput